test/unit/sourceset.test.js

```diff
@@ -0,0 +1,499 @@
+/* eslint-env qunit */
+import videojs from '../../src/js/video.js';
+import document from 'global/document';
+import window from 'global/window';
+
+const testSrc = {
+  src: 'http://vjs.zencdn.net/v/oceans.mp4',
+  type: 'video/mp4'
+};
+
+const wait = 400;
+let qunitFn = 'module';
+
+if (videojs.browser.IS_IE8) {
+  qunitFn = 'skip';
+}
+
+const Html5 = videojs.getTech('Html5');
+const oldMovingMedia = Html5.prototype.movingMediaElementInDOM;
+
+const validateSource = function(assert, player, sources, checkVideoSource = true) {
```

Comment:
All the tests validate that the source is correct, although some have to skip the "video" source, which can technically be empty string/null if we are using a source element.